### ISSUE TITLE
Flickering of depth image/buffer when fps is low and device is rotated

### ISSUE THREAD

**Issue by MoritzSkowronski** [keijiro/Bibcam#11]
Hi,

we are seeing some strange behavior in Bibcam that we can't get to the bottom of. Therefore, I was wondering if you've experienced the same issue:

When the iOS device's fps gets below 20 fps or so and we rotate the device (with the orientation of the device locked!), the depth image "flickers" which results in all pixels getting drawn to something almost planar. Here are two images of the issue, the first one being a correct frame, the second one, where the buffer flickers:

![1 Kopie](https://user-images.githubusercontent.com/12486612/183355339-25455bd4-e7ed-4dad-ba92-49bd82101caf.jpg)

![2 Kopie](https://user-images.githubusercontent.com/12486612/183355362-72fae6b6-76ea-42bd-8ad8-d3b62801bc9f.jpg)
 
PS: We encode the confidence values for each pixels in the green channel of the stencil buffer. This is not causing the issue, it's also happening in a "clean" Bibcam.

**Comment by keijiro** [comment]
I have never experienced such an issue. I'd recommend checking if the issue reproduces with the AR Foundation DepthImages sample.

https://github.com/Unity-Technologies/arfoundation-samples

**Comment by MoritzSkowronski** [comment]
Hey, 

thank you for the quick reply! We are actually a bit smarter now: We had changed the "Environment Depth Mode" in the AR Occlusion Manager from "Best" to "Fastest" earlier in the project. This seems to have some strange side effects that only occur when a) the frame rate drops and b) the devices orientation is changed from its default landscape to any other orientation. I don't know what ARKit does behind the scenes that actually causes this, but by switching back to "Best" we managed to fix the error!
